test/Microsoft.IdentityModel.Tests/IdentityComparer.cs

```diff
@@ -37,108 +37,83 @@
-        /// </summary>
-        /// <param name="context"></param>
-        /// <returns>true if any diffs were added.</returns>
-        public bool Merge(CompareContext context)
+        public static bool AreSecurityKeyEnumsEqual(object o1, object o2, CompareContext context)
         {
-            return Merge(null, context);
+            IEnumerable<SecurityKey> securityKeyEnum1 = (IEnumerable<SecurityKey>)o1;
+            IEnumerable<SecurityKey> securityKeyEnum2 = (IEnumerable<SecurityKey>)o2;
+            return AreEnumsEqual<SecurityKey>(securityKeyEnum1, securityKeyEnum2, context, AreSecurityKeysEqual);
         }
 
-        public bool Merge(string title, CompareContext context)
+        public static bool AreObjectEnumsEqual(object o1, object o2, CompareContext context)
         {
-            if (context == null)
-                return false;
-
-            if (context.Diffs.Count > 0)
-            {
-                _diffs.Add(title ?? string.Empty);
-                _diffs.AddRange(context.Diffs);
-            }
-
-            return (context.Diffs.Count == 0);
+            IEnumerable<object> objectEnum1 = (IEnumerable<object>)o1;
+            IEnumerable<object> objectEnum2 = (IEnumerable<object>)o2;
+            return AreEnumsEqual<object>(objectEnum1, objectEnum2, context, AreObjectsEqual);
         }
 
-        public bool IgnoreClaimsIdentityType { get; set; }
-
-        public bool IgnoreClaimsPrincipalType { get; set; }
-
-        public bool IgnoreClaimType { get; set; }
-
-        public bool IgnoreProperties { get; set; }
-
-        public bool IgnoreSubject { get; set; } = true;
-
-        public bool IgnoreTokenStreamReader { get; set; } = true;
-
-        public bool IgnoreType { get; set; }
-
-        public StringComparison StringComparison { get; set; } = System.StringComparison.Ordinal;
-
-        public string Title { get; set; }
-    }
-
-    public class IdentityComparer
-    {
         public static bool AreEnumsEqual<T>(IEnumerable<T> t1, IEnumerable<T> t2, CompareContext context, Func<T, T, CompareContext, bool> areEqual)
         {
-            if (t1 == null && t2 == null)
-                return true;
+            // Check if either t1 or t2 are null or references of each other to see if we can terminate early.
+            // Is this necessary if we're already checking this in AreEqual()?
+            bool? res = AreObjectsNullOrReferences(t1,t2,context);
 
-            if (t1 == null)
-            {
-                context.Diffs.Add($"t1 == null, t2 != null, (type: {t2.GetType()}, value: {t2.ToString()}).");
-                return false;
-            }
-
-            if (t2 == null)
-            {
-                context.Diffs.Add($"t1 != null, t2 == null, (type: {t1.GetType()}), value: {t1.ToString()}).");
-                return false;
-            }
-
-            if (ReferenceEquals(t1, t2))
-                return true;
+            if (res != null)
```

Comment:
for nullablies use a different pattern:
if (res.HasValue)
    return res.Value;

(change the other instances)